`test_conveyor` consistently failing

There's a consistent issue with test_conveyor. It seems to have begun 2 days ago: https://github.com/rucio/rucio/actions/workflows/integration_tests.yml

I'm not sure if this might be related to the MySQL8 autotest issue: https://github.com/rucio/rucio/issues/6747

These are the failing tests:
```
 =========================== short test summary info ============================
FAILED tests/test_conveyor.py::test_preparer_throttler_submitter[file_config_mock{'conveyor': {'use_preparer': 'true'}}] - AssertionError: assert 1.0 == 0
FAILED tests/test_conveyor.py::test_transfer_plugins[file_config_mock{'transfers': {'fts3tape_metadata_plugins': 'activity'}, 'tape_priority': {'fast': '100', 'slow': '1'}}] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> != <RequestSta...
FAILED tests/test_conveyor.py::test_multihop_intermediate_replica_lifecycle[caches_mock0-core_config_mock{'transfers': {'multihop_tombstone_delay': -1}}] - AssertionError: assert <ReplicaState.COPYING: 'C'> == <ReplicaState.AVAILAB...
FAILED tests/test_conveyor.py::test_fts_non_recoverable_failures_handled_on_multihop[caches_mock0] - TypeError: argument of type 'NoneType' is not iterable
FAILED tests/test_conveyor.py::test_fts_recoverable_failures_handled_on_multihop[caches_mock0] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_receiver_archiving[caches_mock0] - assert False
FAILED tests/test_conveyor.py::test_non_deterministic_dst[caches_mock0] - AssertionError: assert <ReplicaState.COPYING: 'C'> == <ReplicaState.AVAILAB...
FAILED tests/test_conveyor.py::test_lost_transfers - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_overwrite_on_tape[caches_mock0] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_overwrite_hops[caches_mock0] - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='fts', port=8...
FAILED tests/test_conveyor.py::test_file_exists_handled[caches_mock0] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_overwrite_corrupted_files[caches_mock0-core_config_mock{'transfers': {'overwrite_corrupted_files': False}}] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_two_multihops_same_intermediate_rse[file_config_mock{'transfers': {'stats_enabled': 'True'}}-caches_mock0-core_config_mock{'transfers': {'multihop_tombstone_delay': -1, 'multihop_rse_expression': '*'}}] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_checksum_validation - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_multisource_receiver - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_multihop_receiver_on_failure[caches_mock0] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_multihop_receiver_on_success[caches_mock0] - AssertionError: assert <RequestState.SUBMISSION_FAILED: 'A'> == <RequestSta...
FAILED tests/test_conveyor.py::test_cancel_rule - TypeError: sequence item 0: expected str instance, NoneType found
FAILED tests/test_conveyor.py::test_transfer_with_tokens[file_config_mock{'oidc': {'admin_issuer': 'indigoiam'}}] - assert False
FAILED tests/test_conveyor.py::test_stager - AssertionError: assert <ReplicaState.COPYING: 'C'> == <ReplicaState.AVAILAB...
FAILED tests/test_conveyor.py::test_multisource[caches_mock0] - AssertionError: assert 'XRD1' == 'XRD4'
======= 21 failed, 4 passed, 1 skipped, 29 warnings in 865.90s (0:14:25) =======
Error: Process completed with exit code 1.
```

--- a/etc/docker/dev/docker-compose.yml
+++ b/etc/docker/dev/docker-compose.yml
@@ -152,7 +152,7 @@ services:
         soft: 10240
         hard: 10240
   ftsdb:
-    image: docker.io/mysql:8
+    image: docker.io/mysql:8.3
     platform: linux/amd64
     profiles:
       - storage

--- a/etc/docker/dev/docker-compose.yml
+++ b/etc/docker/dev/docker-compose.yml
@@ -105,7 +105,7 @@ services:
       - POSTGRES_PASSWORD=rucio
     command: ["-c", "fsync=off","-c", "synchronous_commit=off","-c", "full_page_writes=off"]
   mysql8:
-    image: docker.io/mysql:8
+    image: docker.io/mysql:8.3
     platform: linux/amd64
     profiles:
       - mysql8

